modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/js/main/ApiClientAsync.kt.mustache

```diff
@@ -120,3 +109,13 @@ public open class ApiClientAsync(
         }
     }
 }
+
+private fun createHttpClient(httpClientEngine: HttpClientEngine? = null, serializer: KotlinxSerializer): HttpClient {
+    val jsonConfig: JsonFeature.Config.() -> Unit = { this.serializer = serializer }
+    val clientConfig: (HttpClientConfig<*>) -> Unit = { it.install(JsonFeature, jsonConfig) }
+    return if (httpClientEngine == null) {
+        HttpClient(clientConfig)
+    } else {
+        HttpClient(httpClientEngine, clientConfig)
+    }
+}
```

Comment:
Maybe we can get some common code here?

Reply:
We can pull make this a common util by extracting or changing function visibility to internal

Reply:
Yeah that's what I was aiming at

Reply:
You could just make the `ApiClient` one internal probably